src/core/public/overlays/sidecar/components/resizable_button.tsx

```diff
@@ -29,6 +29,30 @@ export const ResizableButton = ({ dockedMode, onResize, flyoutSize }: Props) =>
   const initialFlyoutSize = useRef(flyoutSize);
   const setFocus = (e: React.MouseEvent<HTMLButtonElement>) => e.currentTarget.focus();
 
+  useEffect(() => {
+    const handleWindowResize = () => {
+      if (flyoutSize > MIN_SIDECAR_SIZE) {
+        // Make sure flyout never below min size even if the window goes below the size
+        if (dockedMode === SIDECAR_DOCKED_MODE.TAKEOVER && flyoutSize > window.innerHeight) {
+          // Automatically reduce the height in full screen mode when resize the window
+          onResize(window.innerHeight);
+        } else if (
+          (dockedMode === SIDECAR_DOCKED_MODE.LEFT || dockedMode === SIDECAR_DOCKED_MODE.RIGHT) &&
+          flyoutSize > window.innerWidth
+        ) {
+          // Automatically reduce the width in left or right docked mode when resize the window
+          onResize(window.innerWidth);
+        }
+      }
+    };
+
+    window.addEventListener('resize', handleWindowResize);
```

Comment:
Can we optimize this listener registration logic? It will be call many times after `flyoutSize` updated.

Reply:
cool, I will add debouncing here